Content.Shared/Gibbing/Systems/GibbingSystem.cs

```diff
@@ -0,0 +1,309 @@
+                        }
+                    }
+                    break;
+                }
+        }
+
+        switch (gibType)
+        {
+            case GibType.Skip:
+                break;
+            case GibType.Drop:
+            {
+                DropEntity(target, parentXform, randomSpreadMod, gibable, ref droppedEntities, launchGibs,
+                    launchDirection, launchImpulse, launchImpulseVariance, launchCone);
+                break;
+            }
+            case GibType.Gib:
+            {
+                GibEntity(target, parentXform, randomSpreadMod, gibable, ref droppedEntities, launchGibs,
+                    launchDirection, launchImpulse, launchImpulseVariance, launchCone);
+                break;
+            }
+        }
+        if (playAudio)
+            _audioSystem.PlayPredicted(gibable.GibSound, parentXform.Coordinates, null, GibbableComponent.GibAudioParams);
+
+        if (gibType == GibType.Gib)
+            EntityManager.QueueDeleteEntity(target);
+        return true;
+    }
+
+    private void DropEntity(EntityUid target, TransformComponent parentXform, float randomSpreadMod, GibbableComponent? gibable,
+        ref HashSet<EntityUid> droppedEntities, bool flingEntity, Vector2? scatterDirection, float scatterImpulse,
+        float scatterImpulseVariance, Angle scatterCone)
+    {
+        var gibCount = 0;
+        if (Resolve(target, ref gibable, logMissing: false))
+        {
+            gibCount = gibable.GibCount;
+        }
+        var gibAttemptEvent = new AttemptEntityGibEvent(target, gibCount, GibType.Drop);
+        RaiseLocalEvent(target, ref gibAttemptEvent);
+        switch (gibAttemptEvent.GibType)
+        {
+            case GibType.Skip:
+                return;
+            case GibType.Gib:
+                GibEntity(target, parentXform, randomSpreadMod, gibable, ref droppedEntities, flingEntity,scatterDirection,
+                    scatterImpulse, scatterImpulseVariance, scatterCone ,deleteTarget: false);
+                return;
+        }
+        _transformSystem.AttachToGridOrMap(target);
+        _transformSystem.SetCoordinates(target,parentXform.Coordinates);
+        _transformSystem.SetWorldRotation(target, _random.NextAngle());
+        droppedEntities.Add(target);
+        if (flingEntity)
+        {
+            FlingDroppedEntity(target, scatterDirection, scatterImpulse, scatterImpulseVariance, scatterCone);
+        }
+        var gibbedEvent = new EntityGibbedEvent(target, new List<EntityUid>{target});
+        RaiseLocalEvent(target,ref gibbedEvent);
+    }
+
+    private List<EntityUid> GibEntity(EntityUid target, TransformComponent parentXform, float randomSpreadMod,
+        GibbableComponent? gibable, ref HashSet<EntityUid> droppedEntities, bool flingEntity, Vector2? scatterDirection, float scatterImpulse,
+        float scatterImpulseVariance, Angle scatterCone, bool deleteTarget = true)
+    {
+        var localGibs = new List<EntityUid>();
```

Comment:
store this list on the system and just clear it here, microoptimisation

Reply:
Storing data on the system directly would cause issues if we want to do update slicing or threading in the future.